index-protocol/src/Ergvein/Index/Protocol/Deserialization.hs

```diff
@@ -61,6 +62,22 @@ word8toFeeLevel = \case
   2 -> Just FeeCheap
   _ -> Nothing
 
+versionParser :: Parser ProtocolVersion
+versionParser = do
+  bs :: S.Bitstream (S.Right) <- S.fromBits <$> anyWord32be
+  let pref = S.take i2 bs
+  let rst  = S.drop i2 bs
+  let mj   = S.toBits $ S.append pad $ S.take i10 rst
+  let mn   = S.toBits $ S.append pad $ S.take i10 $ S.drop i10 rst
+  let p    = S.toBits $ S.append pad $ S.take i10 $ S.drop i20 rst
+  if (pref == protocolReservedBits)
```

Comment:
Lets ignore reserved bits content to allow soft fork if we need them in future instead of hard fork